src/zodiaq/identification/matchingFunctions.py

```diff
@@ -61,11 +65,10 @@ def match_query_peak_to_all_succeeding_library_peaks_within_tolerance(
 
 @njit
 def numba_enhanced_matching_of_library_to_query_pooled_spectra(
-    libraryPeaks, queryPeaks, ppmTolerance
+    libraryPeaks, queryPeaks, ppmTolerance, baselineLibraryIdx, baselineQueryIdx
 ):
-    baselineLibraryIdx, baselineQueryIdx = np.int64(0), np.int64(0)
     mzIdx, intensityIdx, tagIdx = 0, 1, 2
-    rowNum = 8000
+    rowNum = 1000000
     data = np.empty(shape=(rowNum, 6))
```

Comment:
I don't think that pre-allocation is giving you a performance benefit here; since the entries are objects and you don't know the final number of entries, it might be better to keep this is a list and append to the end of it.

Reply:
It's a numba-specific alteration - with functions with the @njit decorator this significantly impacts the speed of the program. In normal python code, I would completely agree.